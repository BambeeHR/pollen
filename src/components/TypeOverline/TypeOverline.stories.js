import Typography from '../../constants/Typography';
import TypeOverline from './TypeOverline.vue';

const { Overline } = Typography;

export default {
  title: 'TypeOverline',
  component: TypeOverline,
};

export const Gallery = () => ({
  components: { TypeOverline },
  data() {
    return {
      variants: Object.values(Overline),
    };
  },
  template: `
    <div>
      <div v-for="variant in variants" :key="variant" :style="{ marginBottom: '48px' }">
        <TypeOverline tag="h1">{{ variant }}</TypeOverline>
        <TypeOverline 
          :variant="variant"
          tag="div"
        >
          The quick brown fox jumps over the lazy dog.
        </TypeOverline>
      </div>
    </div>
  `,
});
