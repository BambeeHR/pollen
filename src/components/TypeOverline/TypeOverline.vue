<template functional>
  <component
    :is="props.tag"
    :class="[`type-overline-${props.variant}`, data.class, data.staticClass]"
    class="type-overline"
    v-bind="$options.getAttributes({ data, parent })"
    v-on="listeners"
  >
    <slot />
  </component>
</template>

<script>
import Typography from '../../constants/Typography';
import getAttributes from '../../util/getAttributes';

const { Overline } = Typography;

/**
 * A styled block of text.
 */
export default {
  props: {
    /** The html tag to use for this component. */
    tag: {
      type: String,
      default: 'span',
    },
    /** One of 'small', 'medium', or 'large'. */
    variant: {
      type: String,
      default: Typography.Overline.SMALL,
      validator: (value) => Object.values(Overline).includes(value),
    },
  },
  getAttributes,
};
</script>

<style>
/* Intentionally left unscoped to provide global helper styles. */
@each $variant in small, medium, large {
  .type-overline-$(variant) {
    @apply font-overline-$(variant);
  }
}
</style>
