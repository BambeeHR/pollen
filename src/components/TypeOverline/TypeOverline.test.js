import { mount } from '@vue/test-utils';
import TypeOverline from './TypeOverline.vue';

describe('TypeOverline', () => {
  test('renders correctly', () => {
    const wrapper = mount(TypeOverline, {
      slots: {
        default: 'Hello world!',
      },
    });
    expect(wrapper.element).toMatchSnapshot();
  });
});
