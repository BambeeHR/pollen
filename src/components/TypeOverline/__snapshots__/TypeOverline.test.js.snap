// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TypeOverline renders correctly 1`] = `
<span
  class="type-overline type-overline-small"
>
  Hello world!
</span>
`;
